Source/JavaScriptCore/inspector/remote/cocoa/RemoteConnectionToTargetCocoa.mm

```diff
@@ -215,9 +215,10 @@ static void RemoteTargetHandleRunSourceWithInfo(void* info)
                 target->disconnect(*this);
 
             m_target = nullptr;
-            if (targetIdentifier)
-                RemoteInspector::singleton().updateTargetListing(targetIdentifier);
```

Comment:
i think this change should also be made in `Source/JavaScriptCore/inspector/remote/RemoteConnectionToTarget.cpp`

Reply:
Updated the C++ implementation to match the Cocoa one.

Reply:
<img width="680" alt="Screenshot 2025-03-11 at 10 28 21 AM" src="https://github.com/user-attachments/assets/84f7e242-37d6-45a5-a368-c02ff33ab85f" />

I think it would be enough to add `if (targetIdentifier)` to the existing `updateTargetListing`.

Reply:
But wouldn't `RefPtr target =m_target.get()` increase the reference count and thus keep the target alive for the duration of the block scope? 

Wrapping that into its own scope and separating it from the call to  `RemoteInspector::singleton().updateTargetListing` is what the solution is for the Cocoa implementation.